assets/js/googlesitekit/notifications/register-defaults.js

```diff
@@ -221,8 +244,14 @@ export function registerDefaults( notificationsAPI ) {
 			const viewOnly =
 				SITE_KIT_VIEW_ONLY_CONTEXTS.includes( viewContext );
 
-			const isAnalyticsConnected =
-				select( CORE_MODULES ).isModuleConnected( 'analytics-4' );
+			const isAnalyticsConnected = await resolveSelect(
+				CORE_MODULES
+			).isModuleConnected( 'analytics-4' );
+
+			if ( viewOnly ) {
+				// This ensures resolution for the canViewSharedModule() selectors.
+				await resolveSelect( CORE_MODULES ).getModules();
+			}
```

Comment:
We can call `await resolveSelect( CORE_MODULES ).getModules();` unconditionally here because `getModules` will be resolved in `isModuleConnected` anyway. Plus, we need to resolve `isGatheringData` and `getRecoverableModules`:

```js
const viewOnly = SITE_KIT_VIEW_ONLY_CONTEXTS.includes( viewContext );

await Promise.all( [
	resolveSelect( CORE_MODULES ).getModules(),
	resolveSelect( MODULES_ANALYTICS_4 ).isGatheringData(),
	viewOnly ? resolveSelect( CORE_MODULES ).getRecoverableModules() : Promise.resolve( [] ),
] );

...

const isAnalyticsConnected = select( CORE_MODULES ).isModuleConnected( 'analytics-4' );
```